module/assets/helper.js

```diff
@@ -0,0 +1,353 @@
+// Dependencies
+const programsHelper = require(MODULES_BASE_PATH + "/programs/helper");
+const solutionsHelper = require(MODULES_BASE_PATH + "/solutions/helper");
+const userExtensionsHelper = require(MODULES_BASE_PATH +
+  "/user-extension/helper");
+const userRolesHelper = require(MODULES_BASE_PATH + "/user-roles/helper");
+const kafkaProducersHelper = require(ROOT_PATH + "/generics/kafka/producers");
+
+module.exports = class AssetsHelper {
+  static fetchPrograms(queryData, bodyData) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let organizationAssets;
+
+        switch (queryData) {
+          case "program":
+            organizationAssets =
+              await programsHelper.queryForOrganizationPrograms(
+                bodyData,
+                queryData
+              );
+            break;
+          case "solution":
+            organizationAssets =
+              await solutionsHelper.queryForOrganizationSolutions(bodyData);
+
+            break;
+          default:
+            let allOrganizationProgram =
+              await programsHelper.queryForOrganizationPrograms(
+                bodyData,
+                queryData
+              );
+            let allOrganizationSolutions =
+              await solutionsHelper.queryForOrganizationSolutions(bodyData);
+            organizationAssets = [
+              ...allOrganizationProgram.data,
+              ...allOrganizationSolutions.data,
+            ];
+
+            break;
+        }
+
+        if (queryData !== "" && queryData !== undefined) {
+          return resolve({
+            result: organizationAssets,
+          });
+        } else {
+          return resolve({
+            success: true,
+            message: constants.apiResponses.ASSETS_SUCCESSFULLY,
+            result: organizationAssets,
+          });
+        }
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  static ownershipTransfer(ownershipTransferEvent) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let reqData = ownershipTransferEvent.edata;
+        let fromFindQuery = {
+          userId: reqData.fromUserProfile.userId,
+          [programRoles]: { $exists: true },
+        };
+        let fromUpdateQuery = {
+          userId: reqData.fromUserProfile.userId,
+          $unset: { programRoles: [] },
+        };
+        let toFindQuery = {
+          userId: reqData.toUserProfile.userId,
+          [programRoles]: { $exists: true },
+        };
+        const newCollectionForUserExtension = {
+          roles: [],
+          status: "active",
+          isDeleted: false,
+          removedFromHomeScreen: [],
+          improvementProjects: [],
+          platformRoles: [
+            {
+              roleId: userRoleId,
```

Comment:
from where you are getting this userRoleId @praveenKDass

Reply:
resolved